src/Box/Box.ts

```diff
@@ -0,0 +1,51 @@
+import styled from '@emotion/styled';
+import {
+  borders,
+  BordersProps,
+  boxShadow,
+  BoxShadowProps,
+  display,
+  DisplayProps,
+  gridArea,
+  GridAreaProps,
+  height,
+  HeightProps,
+  maxHeight,
+  minHeight,
+  position,
+  PositionProps,
+  width,
+  WidthProps
+} from 'styled-system';
+
+import { PolyComponent } from '../polyComponent';
+
+import Text, { TextProps } from '../Text';
+import { WithColorPropFix } from '../colorPropFix';
+
+export type Props = TextProps &
+  BordersProps &
+  BoxShadowProps &
+  DisplayProps &
+  HeightProps &
+  GridAreaProps &
+  PositionProps &
+  WidthProps;
+
+const TextDiv = Text.withComponent('div');
+const TextDivWithColorPropFix: WithColorPropFix<typeof TextDiv> = TextDiv;
+
+const Box: PolyComponent<'div', Props> = styled(TextDivWithColorPropFix)<Props>`
```

Comment:
`WithColorPropFix` is a solution to a conflict between React's and Styled-System's `color` props; see `src/colorPropFix.ts`.  I hope to simplify this later as it's somewhat awkward to use right now.

`PolyComponent` is a solution for components with `as` props to accept the right `props` for the element type.  See `src/polyComponent.ts`.